internal/backend/sessions/sessioncalls/calls.go

```diff
@@ -58,6 +59,7 @@ type calls struct {
 	//                 user script.
 	generalWorker worker.Worker
 	uniqueWorker  worker.Worker
+	mu            *sync.Mutex
```

Comment:
i think we need to use a RWMutex, and protect also the reads?

Reply:
also i would call it sessionsMutex, and since executorsForSessions is global it needs to be in the global scope.